invalid memory address or nil pointer dereference

when playing with the tool i got this error after about 500 requests.
Unfortunately the tools does not tell me which request caused the issue.

```
INFO  wiretap is proxying all traffic to 'https://xxxxxxxxxxx'

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x68 pc=0xa4ddca]

goroutine 11173 [running]:
github.com/pb33f/libopenapi-validator/requests.(*requestBodyValidator).ValidateRequestBody(0xc000da9d00, 0xc0004a7b00)
        /go/pkg/mod/github.com/pb33f/libopenapi-validator@v0.0.34/requests/validate_body.go:32 +0x16a
github.com/pb33f/libopenapi-validator.(*validator).ValidateHttpRequest.func2(0xc000f986c0?, 0x0?)
        /go/pkg/mod/github.com/pb33f/libopenapi-validator@v0.0.34/validator.go:247 +0x30
created by github.com/pb33f/libopenapi-validator.(*validator).ValidateHttpRequest in goroutine 11186
        /go/pkg/mod/github.com/pb33f/libopenapi-validator@v0.0.34/validator.go:267 +0x346
```

--- a/go.mod
+++ b/go.mod
@@ -3,13 +3,13 @@ module github.com/pb33f/wiretap
 go 1.21
 
 require (
-	github.com/google/uuid v1.4.0
+	github.com/google/uuid v1.5.0
 	github.com/lucasjones/reggen v0.0.0-20200904144131-37ba4fa293bb // indirect
-	github.com/pb33f/libopenapi v0.13.21
-	github.com/pb33f/libopenapi-validator v0.0.34
+	github.com/pb33f/libopenapi v0.14.4
+	github.com/pb33f/libopenapi-validator v0.0.37
 	github.com/pb33f/ranch v0.4.0
-	github.com/pterm/pterm v0.12.71
-	golang.org/x/exp v0.0.0-20231206192017-f3f8817b8deb // indirect
+	github.com/pterm/pterm v0.12.72
+	golang.org/x/exp v0.0.0-20231219180239-dc181d75b848 // indirect
 )
 
 require (
@@ -49,9 +49,9 @@ require (
 	github.com/vmware-labs/yaml-jsonpath v0.3.2 // indirect
 	github.com/xo/terminfo v0.0.0-20220910002029-abceb7e1c41e // indirect
 	golang.org/x/sync v0.1.0 // indirect
-	golang.org/x/sys v0.14.0 // indirect
-	golang.org/x/term v0.13.0 // indirect
-	golang.org/x/text v0.13.0 // indirect
+	golang.org/x/sys v0.15.0 // indirect
+	golang.org/x/term v0.15.0 // indirect
+	golang.org/x/text v0.14.0 // indirect
 	gopkg.in/ini.v1 v1.62.0 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1

--- a/mock/mock_engine.go
+++ b/mock/mock_engine.go
@@ -45,7 +45,7 @@ func (rme *ResponseMockEngine) GenerateResponse(request *http.Request) ([]byte, 
 
 func (rme *ResponseMockEngine) ValidateSecurity(request *http.Request, operation *v3.Operation) error {
 	// get out early if there is nothing to do.
-	if len(rme.doc.Components.SecuritySchemes) <= 0 {
+	if rme.doc.Components.SecuritySchemes.Len() <= 0 {
 		return nil
 	}
 
@@ -54,7 +54,9 @@ func (rme *ResponseMockEngine) ValidateSecurity(request *http.Request, operation
 	// global security
 	if len(rme.doc.Security) > 0 {
 		for _, securityRequirement := range rme.doc.Security {
-			for key, scopes := range securityRequirement.Requirements {
+			for securityPairs := securityRequirement.Requirements.First(); securityPairs != nil; securityPairs = securityPairs.Next() {
+				key := securityPairs.Key()
+				scopes := securityPairs.Value()
 				mustApply[key] = scopes
 			}
 		}
@@ -63,7 +65,9 @@ func (rme *ResponseMockEngine) ValidateSecurity(request *http.Request, operation
 	// operation security
 	if len(operation.Security) > 0 {
 		for _, securityRequirement := range operation.Security {
-			for key, scopes := range securityRequirement.Requirements {
+			for securityPairs := securityRequirement.Requirements.First(); securityPairs != nil; securityPairs = securityPairs.Next() {
+				key := securityPairs.Key()
+				scopes := securityPairs.Value()
 				mustApply[key] = scopes
 			}
 		}
@@ -75,7 +79,7 @@ func (rme *ResponseMockEngine) ValidateSecurity(request *http.Request, operation
 
 		for scope, _ := range mustApply {
 
-			securityComponent := rme.doc.Components.SecuritySchemes[scope]
+			securityComponent := rme.doc.Components.SecuritySchemes.GetOrZero(scope)
 			if securityComponent != nil {
 
 				// check if we have a security scheme that matches the type.
@@ -145,8 +149,8 @@ func (rme *ResponseMockEngine) findOperation(request *http.Request, pathItem *v3
 		return nil
 	}
 	ops := pathItem.GetOperations()
-	if len(ops) > 0 {
-		op := pathItem.GetOperations()[strings.ToLower(request.Method)]
+	if ops.Len() > 0 {
+		op := pathItem.GetOperations().GetOrZero(strings.ToLower(request.Method))
 		return op
 	}
 	return nil
@@ -301,8 +305,9 @@ func (rme *ResponseMockEngine) runWorkflow(request *http.Request) ([]byte, int, 
 
 func (rme *ResponseMockEngine) findLowestSuccessCode(operation *v3.Operation) string {
 	var lowestCode = 299
-	for key := range operation.Responses.Codes {
-		code, _ := strconv.Atoi(key)
+
+	for codePairs := operation.Responses.Codes.First(); codePairs != nil; codePairs = codePairs.Next() {
+		code, _ := strconv.Atoi(codePairs.Key())
 		if code < lowestCode && code >= 200 {
 			lowestCode = code
 		}
@@ -323,23 +328,23 @@ func (rme *ResponseMockEngine) lookForResponseCodes(
 	// check if the media type exists in the response.
 	for _, code := range resultCodes {
 
-		resp := op.Responses.Codes[code]
+		resp := op.Responses.Codes.GetOrZero(code)
 		if resp == nil {
 			continue
 		}
-		responseBody := resp.Content[mediaTypeString]
+		responseBody := resp.Content.GetOrZero(mediaTypeString)
 		if responseBody != nil {
 			// try and extract a default JSON response
 			return responseBody, false
 		} else {
-			responseBody = resp.Content["application/json"]
+			responseBody = resp.Content.GetOrZero("application/json")
 			return responseBody, false
 		}
 	}
 
 	if op.Responses.Default != nil && op.Responses.Default.Content != nil {
-		if op.Responses.Default.Content[mediaTypeString] != nil {
-			return op.Responses.Default.Content[mediaTypeString], false
+		if op.Responses.Default.Content.GetOrZero(mediaTypeString) != nil {
+			return op.Responses.Default.Content.GetOrZero(mediaTypeString), false
 		}
 	}
 

--- a/mock/mock_engine_test.go
+++ b/mock/mock_engine_test.go
@@ -129,8 +129,8 @@ func TestNewMockEngine_ValidateSecurity_FailAPIKey_Query(t *testing.T) {
 	operation := me.findOperation(request, path)
 
 	// mutate securityScheme to be a query param
-	doc.Components.SecuritySchemes["ApiKeyAuth"].In = "query"
-	doc.Components.SecuritySchemes["ApiKeyAuth"].Name = "pizza-cake-burger"
+	doc.Components.SecuritySchemes.GetOrZero("ApiKeyAuth").In = "query"
+	doc.Components.SecuritySchemes.GetOrZero("ApiKeyAuth").Name = "pizza-cake-burger"
 
 	err := me.ValidateSecurity(request, operation)
 	assert.Error(t, err)
@@ -147,8 +147,8 @@ func TestNewMockEngine_ValidateSecurity_PassAPIKey_Query(t *testing.T) {
 	operation := me.findOperation(request, path)
 
 	// mutate securityScheme to be a query param
-	doc.Components.SecuritySchemes["ApiKeyAuth"].In = "query"
-	doc.Components.SecuritySchemes["ApiKeyAuth"].Name = "pizza-burger-cake"
+	doc.Components.SecuritySchemes.GetOrZero("ApiKeyAuth").In = "query"
+	doc.Components.SecuritySchemes.GetOrZero("ApiKeyAuth").Name = "pizza-burger-cake"
 
 	err := me.ValidateSecurity(request, operation)
 	assert.NoError(t, err)
@@ -164,8 +164,8 @@ func TestNewMockEngine_ValidateSecurity_FailAPIKey_Cookie(t *testing.T) {
 	operation := me.findOperation(request, path)
 
 	// mutate securityScheme to be a query param
-	doc.Components.SecuritySchemes["ApiKeyAuth"].In = "cookie"
-	doc.Components.SecuritySchemes["ApiKeyAuth"].Name = "burger-chips-beer"
+	doc.Components.SecuritySchemes.GetOrZero("ApiKeyAuth").In = "cookie"
+	doc.Components.SecuritySchemes.GetOrZero("ApiKeyAuth").Name = "burger-chips-beer"
 
 	err := me.ValidateSecurity(request, operation)
 	assert.Error(t, err)
@@ -186,8 +186,8 @@ func TestNewMockEngine_ValidateSecurity_PassAPIKey_Cookie(t *testing.T) {
 	operation := me.findOperation(request, path)
 
 	// mutate securityScheme to be a query param
-	doc.Components.SecuritySchemes["ApiKeyAuth"].In = "cookie"
-	doc.Components.SecuritySchemes["ApiKeyAuth"].Name = "burger-chips-beer"
+	doc.Components.SecuritySchemes.GetOrZero("ApiKeyAuth").In = "cookie"
+	doc.Components.SecuritySchemes.GetOrZero("ApiKeyAuth").Name = "burger-chips-beer"
 
 	err := me.ValidateSecurity(request, operation)
 	assert.NoError(t, err)
@@ -204,8 +204,8 @@ func TestNewMockEngine_ValidateSecurity_FailHTTP_Bearer(t *testing.T) {
 	operation := me.findOperation(request, path)
 
 	// mutate securityScheme to be a query param
-	doc.Components.SecuritySchemes["ApiKeyAuth"].Type = "http"
-	doc.Components.SecuritySchemes["ApiKeyAuth"].Scheme = "bearer"
+	doc.Components.SecuritySchemes.GetOrZero("ApiKeyAuth").Type = "http"
+	doc.Components.SecuritySchemes.GetOrZero("ApiKeyAuth").Scheme = "bearer"
 
 	err := me.ValidateSecurity(request, operation)
 	assert.Error(t, err)
@@ -224,8 +224,8 @@ func TestNewMockEngine_ValidateSecurity_PassHTTP_Bearer(t *testing.T) {
 	operation := me.findOperation(request, path)
 
 	// mutate securityScheme to be a query param
-	doc.Components.SecuritySchemes["ApiKeyAuth"].Type = "http"
-	doc.Components.SecuritySchemes["ApiKeyAuth"].Scheme = "bearer"
+	doc.Components.SecuritySchemes.GetOrZero("ApiKeyAuth").Type = "http"
+	doc.Components.SecuritySchemes.GetOrZero("ApiKeyAuth").Scheme = "bearer"
 
 	err := me.ValidateSecurity(request, operation)
 	assert.NoError(t, err)

